src/vos/vos_pool.c

```diff
@@ -1729,22 +1824,29 @@ vos_pool_open_metrics(const char *path, uuid_t uuid, unsigned int flags, void *m
 		goto out;
 	}
 
-	if (!skip_uuid_check && uuid_compare(uuid, pool_df->pd_id)) {
+	if (!(flags & VOS_POF_SKIP_UUID_CHECK) && uuid_compare(uuid, pool_df->pd_id)) {
 		D_ERROR("Mismatch uuid, user="DF_UUIDF", pool="DF_UUIDF"\n",
 			DP_UUID(uuid), DP_UUID(pool_df->pd_id));
 		rc = -DER_ID_MISMATCH;
 		goto out;
 	}
 
-	rc = pool_open(ph, pool_df, flags, metrics, poh);
-	ph = NULL;
-
 out:
+	rc = pool_open_post(&ph, pool_df, flags, metrics, pool, rc);
+	if (rc == 0)
+		*poh = vos_pool2hdl(pool);
+
 	/* Close this local handle, if it hasn't been consumed nor already
 	 * been closed by pool_open upon error.
 	 */
-	if (ph != NULL)
+	if (ph != NULL) {
 		vos_pmemobj_close(ph);
+		ph = NULL;
+	}
+
+	if (unlikely(rc == -DER_EXIST))
+		goto again;
```

Comment:
I don't quite follow this.

Reply:
This is redundant changes left from former modification. I will remove it.